### PR TITLE
Fix ValueError when instantiating SparseTermSimilarityMatrix

### LINKED ISSUE #2688: LeveshteinSimilarityIndex fails when called from SparseTermSimilarityMatrix
#### Problem description

When trying to build a SparseTermSimilarityMatrix using Levenshtein it fails.

#### Steps/code/corpus to reproduce

```python
index = LevenshteinSimilarityIndex(dictionary)
SparseTermSimilarityMatrix(index, dictionary) # <- fails here
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-59-c16b89564835> in <module>
----> 1 similarity_matrix1 = SparseTermSimilarityMatrix(similarity_index1, dictionary)

~/.local/share/virtualenvs/pdftagger-LHy_2RHk/lib/python3.6/site-packages/gensim/similarities/termsim.py in __init__(self, source, dictionary, tfidf, symmetric, positive_definite, nonzero_limit, dtype)
    234                 for term, similarity in index.most_similar(t1, topn=num_rows)
    235                 if term in dictionary.token2id
--> 236             ] if num_rows > 0 else []
    237 
    238             if tfidf is None:

~/.local/share/virtualenvs/pdftagger-LHy_2RHk/lib/python3.6/site-packages/gensim/similarities/levenshtein.py in most_similar(self, t1, topn)
    151             if similarity > 0
    152         )
--> 153         return islice(most_similar, topn)

ValueError: Stop argument for islice() must be None or an integer: 0 <= x <= sys.maxsize.
```

#### Versions

Linux-4.18.0-25-generic-x86_64-with-debian-buster-sid
Python 3.6.9 (default, Nov 18 2019, 15:20:23) 
[GCC 8.3.0]
NumPy 1.17.4
SciPy 1.3.3
gensim 3.8.1
FAST_VERSION 1

### PR DESCRIPTION
**Pull request by ptorrestr**
Force python int before calling `islice`. `topn` is `np.int64` when `LeveshteinSimilarityIndex` is called from `SparseTermSimilarityMatrix`

- fix #2688

### CODE REVIEW

**Comment by piskvorky** [comment]
That looks serious, thanks. Are there any other places that might need the same treatment?

Also, please use a more descriptive ticket title. Keep in mind PR titles become a part of release notes! "fixes XYZ" belongs in the PR description, not the title.

**Comment by ptorrestr** [comment]
So, the error is only happening with python 3.5 and 3.6. It seems that `islice` in python 3.7 is accepting numpy integers.  Not sure if this fix was done in python itself or numpy?

Not sure why Windows builds are failing? Any ideas

**Comment by mpenkov** [comment]
@Please merge gensim/develop into your branch to fix the appveyor issues.

File: gensim/test/test_similarities.py [hunk 1]
```diff
@@ -1225,6 +1225,11 @@ def test_most_similar(self):
         second_similarities = numpy.array([similarity for term, similarity in index.most_similar(u"holiday", topn=10)])
         self.assertTrue(numpy.allclose(first_similarities ** 2.0, second_similarities))
 
+        # check propert integration with SparseTermSimilarityMatrix
```
Comment (mpenkov) [inline gensim/test/test_similarities.py]:
```suggestion
        # check proper integration with SparseTermSimilarityMatrix
```

Comment (mpenkov) [reply gensim/test/test_similarities.py]: Found a typo. I tried to merge this in, but looks like "allow edit from maintainers" is not enabled.

**Comment by mpenkov** [comment]
Merged. Thank you for your effort and your patience @ptorrestr !

**Pull request merged**